hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -263,14 +263,14 @@ class HoodieSparkSqlWriterInternal {
       }
     }
 
+    val keyGenerator = HoodieSparkKeyGeneratorFactory.createKeyGenerator(new TypedProperties(hoodieConfig.getProps))
```

Comment:
we can't instantiate key gen in driver due to auto record key generation.

Reply:
Fixed the Auto Key Gen to do lazy init of configs. Lmk if that works